plugins/woocommerce/src/Internal/Orders/OrderActionsRestController.php

```diff
@@ -130,20 +524,95 @@ public function send_order_details( WP_REST_Request $request ) {
 		WC()->mailer()->customer_invoice( $order );
 
 		$user_agent = esc_html( $request->get_header( 'User-Agent' ) );
-		$note       = sprintf(
+		$messages[] = sprintf(
 			// translators: %1$s is the customer email, %2$s is the user agent that requested the action.
 			esc_html__( 'Order details sent to %1$s, via %2$s.', 'woocommerce' ),
 			esc_html( $order->get_billing_email() ),
 			$user_agent ? $user_agent : 'REST API'
 		);
-		$order->add_order_note( $note, false, true );
+
+		foreach ( $messages as $message ) {
+			$order->add_order_note( $message, false, true );
+		}
 
 		// phpcs:disable WooCommerce.Commenting.CommentHooks.MissingSinceComment
 		/** This action is documented in includes/admin/meta-boxes/class-wc-meta-box-order-actions.php */
 		do_action( 'woocommerce_after_resend_order_email', $order, 'customer_invoice' );
 
 		return array(
-			'message' => __( 'Order details email sent to customer.', 'woocommerce' ),
+			'message' => implode( ' ', $messages ),
 		);
 	}
+
+	/**
+	 * Update the billing email of an order when certain conditions are met.
+	 *
+	 * If the order does not already have a billing email, it will be updated. If it does have one, but `$force` is set
+	 * to `true`, it will be updated. Otherwise this will return an error. This can also return an error if the given
+	 * email address is not valid.
+	 *
+	 * @param WC_Order $order The order to update.
+	 * @param string   $email The email address to maybe add to the order.
+	 * @param bool     $force Optional. True to update the order even if it already has a billing email. Default false.
+	 *
+	 * @return string|WP_Error A message upon success, otherwise an error.
+	 */
+	private function maybe_update_billing_email( WC_Order $order, string $email, ?bool $force = false ) {
+		$existing_email = $order->get_billing_email( 'edit' );
+
+		if ( $existing_email === $email ) {
+			return '';
```

Comment:
Returning an empty string here makes the `$messages` array to have two elements (empty string + the actual message) so when imploding we end up with an extra space at the beginning of the message: 👇 
https://github.com/woocommerce/woocommerce/blob/cceffbee664162384398dd262dda53abc3a8a4c8/plugins/woocommerce/src/Internal/Orders/OrderActionsRestController.php#L542-L544
<img width="310" alt="Screenshot 2025-01-27 at 14 45 09" src="https://github.com/user-attachments/assets/93e6d2c3-7c62-4292-ba9c-0c101710ae57" />

Reply:
Oops, I added `array_filter` to `send_email` to address this issue, but forgot to add it to `send_order_details`. Good catch. Fixed in 0c4fad6634